StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkLogin/LinkLoginFooterView.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  LinkLoginFooterView.swift
+//  StripeFinancialConnections
+//
+//  Created by Mat Schmid on 2024-07-25.
+//
+
+import Foundation
+@_spi(STP) import StripeCore
+@_spi(STP) import StripeUICore
+import UIKit
+
+class LinkLoginFooterView: HitTestView {
+    private let aboveCtaText: String
+    private let cta: String
+    private let theme: FinancialConnectionsTheme
+    private let didSelectCta: () -> Void
+    private let didSelectURL: (URL) -> Void
+
+    private lazy var stackView: UIStackView = {
+        let verticalStackView = UIStackView()
+        verticalStackView.axis = .vertical
+        verticalStackView.spacing = 16
+        verticalStackView.isLayoutMarginsRelativeArrangement = true
+        verticalStackView.directionalLayoutMargins = NSDirectionalEdgeInsets(
+            top: 16,
+            leading: 24,
+            bottom: 16,
+            trailing: 24
+        )
+        verticalStackView.addArrangedSubview(aboveCtaLabel)
+        verticalStackView.addArrangedSubview(ctaButton)
+        return verticalStackView
+    }()
+
+    private lazy var aboveCtaLabel: AttributedTextView = {
+        let termsAndPrivacyPolicyLabel = AttributedTextView(
+            font: .label(.small),
+            boldFont: .label(.smallEmphasized),
+            linkFont: .label(.small),
+            textColor: .textDefault,
+            alignCenter: true
+        )
+        termsAndPrivacyPolicyLabel.setText(
+            aboveCtaText,
+            action: didSelectURL
+        )
+        return termsAndPrivacyPolicyLabel
+    }()
+
+    private lazy var ctaButton: StripeUICore.Button = {
+        let saveToLinkButton = Button.primary(theme: theme)
+        saveToLinkButton.title = cta
+        saveToLinkButton.addTarget(self, action: #selector(didSelectCtaButton), for: .touchUpInside)
+        saveToLinkButton.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            saveToLinkButton.heightAnchor.constraint(equalToConstant: 56)
+        ])
+        return saveToLinkButton
+    }()
+
+    init(
+        aboveCtaText: String,
+        cta: String,
+        theme: FinancialConnectionsTheme,
+        didSelectCta: @escaping () -> Void,
+        didSelectURL: @escaping (URL) -> Void
+    ) {
+        self.aboveCtaText = aboveCtaText
+        self.cta = cta
+        self.theme = theme
+        self.didSelectCta = didSelectCta
+        self.didSelectURL = didSelectURL
+        super.init(frame: .zero)
+        addAndPinSubview(stackView)
```

Comment:
this is not critical, but sharing a perspective:

one reason a lot of existing code is constructing all the views inside of `init` is because it avoids state

for example, it seems like, based off the functions of `enableCtaButton` and `setIsLoading`, the only state you actually need to keep track of is `didSelectCta`, and `ctaButton`

if you use `PaneLayoutView.createFooterView`, you might not even need `didSelectCta`

so this maybe can end up in a way where the only instance variable you have is `ctaButton` (to be able to toggle isEnabled/isLoading)

which then makes it clearer on how this class works

right now, if someone looks at the class, its not immediately clear that the only thing that is being interacted/changed is `ctaButton` (isEnabled/isLoading); there's a lot of instance variables (or "state")

not super important in a small class, but if one follows these principles, it becomes much easier to manage/reason about a large code base, or a large 5000 line UIViewController

Reply:
Resolving since I deleted `LinkLoginFooterView`.